File: framework/ecommerce-framework-api/src/main/java/com/sdl/ecommerce/api/model/ProductPrice.java
package com.sdl.ecommerce.api.model;

/**
 * Product Price
 *
 * @author nic
 */
public interface ProductPrice {

    /**
     * Get price in numeric format
     * @return price
     */
    float getPrice();

    /**
     * Get currency
     * @return currency
     */
    String getCurrency(); // TODO: Have some kind of ISO code here???

    /**
     * Get formatted price to be used in presentation views. This can be localized based on current locale.
     * @return formatted price
     */
    String getFormattedPrice();
}

File: dxa.net/ecommerce-framework-dotnet/SDL.ECommerce.Rest/Model/ProductPrice.cs
using SDL.ECommerce.Api.Model;

namespace SDL.ECommerce.Rest.Model
{
    public class ProductPrice : IProductPrice
    {
        public string Currency { get; set; }
        public string FormattedPrice { get; set; }
        public float? Price { get; set; }
    }
}
